### PR TITLE
feat: add example RICH detector demonstrating and testing `Geant4OpticalTrackerAction`

### PR DESCRIPTION
**Pull request by c-dilks**
This PR adds a proximity-focusing RICH example, pulled out from [EPIC](https://github.com/eic/epic) (and originally developed in [ATHENA](https://eicweb.phy.anl.gov/EIC/detectors/athena)). This example is meant to test #967 for usage of the proposed `OpticalTracker` action.

Two tests are included:
1. run a test simulation, throw pions at the RICH at fixed eta and energy.
1. determine the average number of hits, which seems to be about 230; I'm not sure how realistic this number is, but having a test on this number will at least keep it stable with respect to continued DD4hep development

Note: this PR's branch is based on the #967 branch (so it includes the proposed `Geant4OpticalTrackerAction` in the diff)

BEGINRELEASENOTES
- add example RICH detector, demonstrating and testing `Geant4OpticalTrackerAction`

ENDRELEASENOTES

### CODE REVIEW

**Comment by c-dilks** [comment]
(squashed all commits to 1)

**Comment by c-dilks** [comment]
Some `linux` CI jobs are failing during the new test `t_OpticalTracker_simulation` (calls `richsim.py`); segfaults with the following stack trace:
```
2022-09-15T01:33:06.8417343Z ===========================================================
2022-09-15T01:33:06.8417571Z There was a crash.
2022-09-15T01:33:06.8417845Z This is the entire stack trace of all threads:
2022-09-15T01:33:06.8418134Z ===========================================================
2022-09-15T01:33:06.8418412Z #0  0x00007fbe100c560c in waitpid () from /lib64/libc.so.6
2022-09-15T01:33:06.8418853Z #1  0x00007fbe10042f62 in do_system () from /lib64/libc.so.6
2022-09-15T01:33:06.8420244Z #2  0x00007fbe0f35e00c in TUnixSystem::StackTrace() () from /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.24.00-a725e/x86_64-centos7-gcc10-opt/lib/libCore.so
2022-09-15T01:33:06.8421330Z #3  0x00007fbe0f5c7b5a in (anonymous namespace)::TExceptionHandlerImp::HandleException(int) () from /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.24.00-a725e/x86_64-centos7-gcc10-opt/lib/libcppyy_backend3_8.so
2022-09-15T01:33:06.8422187Z #4  0x00007fbe0f35b491 in TUnixSystem::DispatchSignals(ESignals) () from /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.24.00-a725e/x86_64-centos7-gcc10-opt/lib/libCore.so
2022-09-15T01:33:06.8422595Z #5  <signal handler called>
2022-09-15T01:33:06.8422873Z #6  0x00007fbe10036387 in raise () from /lib64/libc.so.6
2022-09-15T01:33:06.8423157Z #7  0x00007fbe10037a78 in abort () from /lib64/libc.so.6
2022-09-15T01:33:06.8423469Z #8  0x00007fbe10078f67 in __libc_message () from /lib64/libc.so.6
2022-09-15T01:33:06.8423781Z #9  0x00007fbe10081329 in _int_free () from /lib64/libc.so.6
2022-09-15T01:33:06.8424175Z #10 0x00007fbdf51588ae in dd4hep::xml::Tag_t::~Tag_t() () from /home/runner/work/DD4hep/DD4hep/lib/libDDCore.so.1.22
2022-09-15T01:33:06.8424560Z #11 0x00007fbe10039ce9 in __run_exit_handlers () from /lib64/libc.so.6
2022-09-15T01:33:06.8424876Z #12 0x00007fbe10039d37 in exit () from /lib64/libc.so.6
2022-09-15T01:33:06.8425617Z #13 0x00007fbe0f3577ba in TUnixSystem::Exit(int, bool) () from /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.24.00-a725e/x86_64-centos7-gcc10-opt/lib/libCore.so
2022-09-15T01:33:06.8426484Z #14 0x00007fbe0f35b491 in TUnixSystem::DispatchSignals(ESignals) () from /cvmfs/sft.cern.ch/lcg/releases/ROOT/v6.24.00-a725e/x86_64-centos7-gcc10-opt/lib/libCore.so
2022-09-15T01:33:06.8426897Z #15 <signal handler called>
2022-09-15T01:33:06.8427181Z #16 0x00007fbe10036387 in raise () from /lib64/libc.so.6
2022-09-15T01:33:06.8427489Z #17 0x00007fbe10037a78 in abort () from /lib64/libc.so.6
2022-09-15T01:33:06.8427795Z #18 0x00007fbe10078f67 in __libc_message () from /lib64/libc.so.6
2022-09-15T01:33:06.8428119Z #19 0x00007fbe10081329 in _int_free () from /lib64/libc.so.6
2022-09-15T01:33:06.8428525Z #20 0x00007fbdf51588ae in dd4hep::xml::Tag_t::~Tag_t() () from /home/runner/work/DD4hep/DD4hep/lib/libDDCore.so.1.22
2022-09-15T01:33:06.8429036Z #21 0x00007fbe10039ce9 in __run_exit_handlers () from /lib64/libc.so.6
2022-09-15T01:33:06.8429355Z #22 0x00007fbe10039d37 in exit () from /lib64/libc.so.6
2022-09-15T01:33:06.8429672Z #23 0x00007fbe1002255c in __libc_start_main () from /lib64/libc.so.6
2022-09-15T01:33:06.8430083Z #24 0x000000000040108e in _start ()
2022-09-15T01:33:06.8430328Z ===========================================================
2022-09-15T01:33:06.8430485Z 
2022-09-15T01:33:06.8430492Z 
2022-09-15T01:33:06.8430641Z The lines below might hint at the cause of the crash.
2022-09-15T01:33:06.8431021Z You may get help by asking at the ROOT forum https://root.cern.ch/forum
2022-09-15T01:33:06.8431413Z Only if you are really convinced it is a bug in ROOT then please submit a
2022-09-15T01:33:06.8431894Z report at https://root.cern.ch/bugs Please post the ENTIRE stack trace
2022-09-15T01:33:06.8432627Z from above as an attachment in addition to anything else
2022-09-15T01:33:06.8436609Z that might help us fixing this issue.
2022-09-15T01:33:06.8436896Z ===========================================================
2022-09-15T01:33:06.8437187Z #16 0x00007fbe10036387 in raise () from /lib64/libc.so.6
2022-09-15T01:33:06.8437499Z #17 0x00007fbe10037a78 in abort () from /lib64/libc.so.6
2022-09-15T01:33:06.8437804Z #18 0x00007fbe10078f67 in __libc_message () from /lib64/libc.so.6
2022-09-15T01:33:06.8438131Z #19 0x00007fbe10081329 in _int_free () from /lib64/libc.so.6
2022-09-15T01:33:06.8438549Z #20 0x00007fbdf51588ae in dd4hep::xml::Tag_t::~Tag_t() () from /home/runner/work/DD4hep/DD4hep/lib/libDDCore.so.1.22
2022-09-15T01:33:06.8438945Z #21 0x00007fbe10039ce9 in __run_exit_handlers () from /lib64/libc.so.6
2022-09-15T01:33:06.8439272Z #22 0x00007fbe10039d37 in exit () from /lib64/libc.so.6
2022-09-15T01:33:06.8439596Z #23 0x00007fbe1002255c in __libc_start_main () from /lib64/libc.so.6
2022-09-15T01:33:06.8439883Z #24 0x000000000040108e in _start ()
2022-09-15T01:33:06.8440125Z ===========================================================
```

**Comment by andresailer** [comment]
Something is causing 
```
2022-09-15T03:20:53.4702573Z 256: TClassTable::Add:0: RuntimeWarning: class dd4hep::NamedObject already in TClassTable
...
```
Probably the components from the DD4hep in the stack are picked up for some reason

File: examples/OpticalTracker/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  REGEX_FAIL "Exception;EXCEPTION;ERROR;Error;FATAL"
+  )
+
+# ---Test: Number of raw photon hits
+dd4hep_add_test_reg( OpticalTracker_number_of_hits
+  COMMAND    "${CMAKE_INSTALL_PREFIX}/bin/run_test_OpticalTracker.sh"
+  EXEC_ARGS  root.exe -b -x -n -q -l
+             "${OpticalTracker_INSTALL}/scripts/test_number_of_hits.C(\"${OpticalTracker_INSTALL}/sim.edm4hep.root\")"
+  REGEX_PASS "TEST: passed"
+  REGEX_FAIL "TEST: failed"
```
Comment (andresailer) [inline examples/OpticalTracker/CMakeLists.txt:57]:
```suggestion
  REGEX_FAIL "TEST: failed"
  DEPENDS OpticalTracker_simulation
```

File: examples/OpticalTracker/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  USES DD4hep::DDCore DD4hep::DDCond ROOT::Core ROOT::Geom ROOT::GenVector ROOT::MathCore)
+install(TARGETS OpticalTrackerExample LIBRARY DESTINATION lib)
+install(DIRECTORY compact scripts DESTINATION ${OpticalTracker_INSTALL} )
+dd4hep_configure_scripts( OpticalTracker DEFAULT_SETUP WITH_TESTS)
+
+# ---Test: run simulation
+dd4hep_add_test_reg( OpticalTracker_simulation
+  COMMAND    "${CMAKE_INSTALL_PREFIX}/bin/run_test_OpticalTracker.sh"
+  EXEC_ARGS  ${Python_EXECUTABLE} ${OpticalTracker_INSTALL}/scripts/richsim.py
+             --outputFile "${OpticalTracker_INSTALL}/sim.edm4hep.root"
```
Comment (andresailer) [inline examples/OpticalTracker/CMakeLists.txt:46]:
```suggestion
             --outputFile "${OpticalTracker_INSTALL}/sim.root"
```
We are not building the EDM4hep plugin for all the CI stacks, so when DD4hep is too old in the stack itself and libDDG4EDM4HEP.so is loaded from the stack and not the CI build of DD4hep things start to crash.

**Review by andresailer — APPROVED** [review]

**Comment by andresailer** [comment]
Thanks!

**Comment by c-dilks** [comment]
Rebased and squashed to 1 commit for cleaner merging, but it looks like that disabled auto-merge.

**Pull request merged**